layerfetcher/layers.go

```diff
@@ -495,7 +362,7 @@ func notifyLayerBlocksResult(layerID types.LayerID, channels []chan LayerPromise
 	hasZeroBlockHash := false
 	var firstErr error
 	for _, res := range peerResult {
-		if res.data != nil {
+		if res.err == nil && res.data != nil {
```

Comment:
could there be a case where both `err` and `data` are nil?

Reply:
from the current impl in `layerBlocksReqReceiver`, no. the code would panic before return `nil, nil`
if the peer does not have data for this layer, it will return `nil, ErrInternal` (with the actual error being database.ErrNotFOund)